src/Console/Commands/InstallAssets.php

```diff
@@ -113,18 +120,50 @@ public static function copyStubs(?array $files = null, bool $force = false, ?\Cl
         }
 
         foreach ($files as $file) {
-            if (file_exists($basePath('tailwind.config.js')) && ! $force) {
+            if (file_exists($basePath($file)) && ! $force && ! $merge) {
                 continue;
             }
 
-            copy(__DIR__ . '/../../../stubs/tailwind/' . $file, $basePath($file));
-            $content = file_get_contents($basePath($file));
+            $merged = false;
+            $oldContent = null;
+            if (file_exists($basePath($file))) {
+                $oldContent = file_get_contents($basePath($file));
+            }
+
+            $content = file_get_contents(__DIR__ . '/../../../stubs/tailwind/' . $file);
             $content = str_replace(
                 '{{ relative_path }}',
                 substr(realpath(__DIR__ . '/../../../'), strlen($basePath())),
                 $content
             );
-            file_put_contents($basePath($file), $content);
+
+            if ($merge && $oldContent) {
+                // if the file is json parsable merge it recursively
+                $decodedContent = json_decode(str_replace('.', '__dot__', $content), true);
+                $decodedOldContent = json_decode(str_replace('.', '__dot__', $oldContent), true);
+
+                if (is_array($decodedContent) && is_array($decodedOldContent)) {
+                    $content = str_replace(
+                        '__dot__',
+                        '.',
+                        json_encode(
+                            Arr::undot(
+                                array_merge(
+                                    Arr::dot($decodedContent),
+                                    Arr::dot($decodedOldContent)
+                                )
+                            ),
+                            JSON_PRETTY_PRINT
+                        )
+                    );
+                }
+
+                $merged = true;
```

Comment:
move this to the if statement above, it is only merged if $content is actually merged.